### PR TITLE
CDAP-14748 MySQL Plugin tests

### PR DESCRIPTION
**Pull request by ibarbashov**

### CODE REVIEW

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlSinkTestRun.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    deployETL(sourceConfig, sinkConfig, "testDBSink");
+    createInputData(inputDatasetName);
+
+
+    try (Connection conn = createConnection();
+         Statement stmt = conn.createStatement()) {
+      stmt.execute("SELECT * FROM my_table");
+      Set<String> users = new HashSet<>();
+      try (ResultSet resultSet = stmt.getResultSet()) {
```
Comment (mlozbin-cybervisiontech) [inline mysql-plugin/src/test/java/co/cask/mysql/MysqlSinkTestRun.java]: stmt.executeQuery()?

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlPostActionTestRun.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      .addPostAction(action)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, config);
+    ApplicationId appId = NamespaceId.DEFAULT.app("postActionTest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+    runETLOnce(appManager, ImmutableMap.of("logical.start.time", "0"));
+
+    try (Connection connection = createConnection()) {
```
Comment (mlozbin-cybervisiontech) [inline mysql-plugin/src/test/java/co/cask/mysql/MysqlPostActionTestRun.java:68]: why not in single try block?

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  }
+
+  @AfterClass
+  public static void tearDownDB() throws SQLException {
+    if (!tearDown) {
+      return;
+    }
+
+    try (Connection conn =
+           createConnection();
```
Comment (mlozbin-cybervisiontech) [inline mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java]: why is it on new line?

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+  protected static void prepareTestData(Connection conn) throws SQLException {
+
+    try (Statement stmt = conn.createStatement()) {
+      stmt.execute("insert into dbActionTest values (1, '1970-01-01')");
+      stmt.execute("insert into postActionTest values (1, '1970-01-01')");
+    }
+
+    try (
+      PreparedStatement pStmt1 =
```
Comment (mlozbin-cybervisiontech) [inline mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java:152]: statements should be in try-with-resource blocks

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlSinkTestRun.java (hunk 1 above)
Comment (ibarbashov) [reply mysql-plugin/src/test/java/co/cask/mysql/MysqlSinkTestRun.java]: done

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlPostActionTestRun.java (hunk 2 above)
Comment (ibarbashov) [reply mysql-plugin/src/test/java/co/cask/mysql/MysqlPostActionTestRun.java:68]: done

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java (hunk 3 above)
Comment (ibarbashov) [reply mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java]: fixed

File: mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java (hunk 4 above)
Comment (ibarbashov) [reply mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java:152]: actually they are

**Pull request closed without merging**